src/tests/ftest/osa/osa_offline_parallel_test.py

```diff
@@ -0,0 +1,261 @@
+        container.create(poh=self.pool.pool.handle,
+                         con_prop=input_param)
+        container.open()
+        obj = DaosObj(self.context, container)
+        obj.create(objcls=1)
+        obj.open()
+        ioreq = IORequest(self.context,
+                          container,
+                          obj, objtype=4)
+        self.test_ioreq = ioreq
+        self.log.info("Writing the Single Dataset")
+        for dkey in range(self.no_of_dkeys):
+            for akey in range(self.no_of_akeys):
+                indata = ("{0}".format(str(akey)[0])
+                          * self.record_length)
+                d_key_value = "dkey {0}".format(dkey)
+                c_dkey = ctypes.create_string_buffer(d_key_value)
+                a_key_value = "akey {0}".format(akey)
+                c_akey = ctypes.create_string_buffer(a_key_value)
+                c_value = ctypes.create_string_buffer(indata)
+                c_size = ctypes.c_size_t(ctypes.sizeof(c_value))
+                ioreq.single_insert(c_dkey, c_akey, c_value, c_size)
+
+    @fail_on(DaosApiError)
+    def verify_single_object(self):
+        """Verify the container data on the existing pool."""
+        self.log.info("Single Dataset Verification -- Started")
+        for dkey in range(self.no_of_dkeys):
+            for akey in range(self.no_of_akeys):
+                indata = ("{0}".format(str(akey)[0]) *
+                          self.record_length)
+                c_dkey = ctypes.create_string_buffer("dkey {0}".format(dkey))
+                c_akey = ctypes.create_string_buffer("akey {0}".format(akey))
+                val = self.ioreq.single_fetch(c_dkey,
+                                              c_akey,
+                                              len(indata)+1)
+                if indata != (repr(val.value)[1:-1]):
+                    self.d_log.error("ERROR:Data mismatch for "
+                                     "dkey = {0}, "
+                                     "akey = {1}".format(
+                                         "dkey {0}".format(dkey),
+                                         "akey {0}".format(akey)))
+                    self.fail("ERROR: Data mismatch for dkey = {0}, akey={1}"
+                              .format("dkey {0}".format(dkey),
+                                      "akey {0}".format(akey)))
+
+    def dmg_thread(self, puuid, rank, target, action, results):
+        """Generate different dmg command related to OSA.
+            Args:
+            puuid (int) : Pool UUID
+            rank (int)  : Daos Server Rank
+            target (list) : Target list
+            action (string) : String to identify OSA action
+                              like drain, reintegration, extend
+            results (queue) : dmg command output queue.
+        """
+        # dmg = DmgCommand(os.path.join(self.prefix, "bin"))
+        dmg = copy.copy(self.dmg_command)
+        self.log.info("Action: {0}".format(action))
```

Comment:
(pylint-logging-format-interpolation) Use % formatting in logging functions and pass the % parameters as arguments